### PR TITLE
CONSOLE-3392: Implement exact search  for more precise result sets and add user prefs setting

### PR DESCRIPTION
**Pull request by jcaianirh**
Fixes: https://issues.redhat.com/browse/CONSOLE-3392

https://github.com/openshift/console/assets/35978579/74a16170-0bbe-47e5-9e5a-d1b9329bb427

### CODE REVIEW

**Review by sg00dwin — APPROVED** [review]

File: frontend/packages/console-app/console-extensions.json
Comment (jhadvig) [inline frontend/packages/console-app/console-extensions.json:434]: I would leave out the `disable` part
```suggestion
      "description": "%console-app~Enable exact string match filtering on resources.%",
```

but not a native speaker so I would rather defer to @opayne1

**Review by jhadvig — APPROVED** [review]
Nice job @jcaianirh :+1
One nit otherwise LGTM
/lgtm

File: frontend/packages/console-app/console-extensions.json
Comment (opayne1) [reply frontend/packages/console-app/console-extensions.json]: I'd say remove disable to stay consistent with the rest of the PR. We also should [refrain from using "/"](https://www.ibm.com/docs/en/ibm-style?topic=punctuation-slashes) because it can be ambiguous. Thanks for looping me in!

Comment (jcaianirh) [reply frontend/packages/console-app/console-extensions.json]: @jhadvig @opayne1 removed the /disable from the description text.

**Comment by opayne1** [comment]
Thanks for the quick demo! 
/label docs-approved

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
/lgtm cancel
The filtering is not applying for all the list pages

**Comment by jcaianirh** [comment]
QE Approver:
 /assign @XiyunZhao
/assign @yapei
PX Approver:
/assign @RickJWagner
Code Approval:
/assign @jhadvig

**Review by RickJWagner — COMMENTED** [review]
Thank you for fulfilling this customer RFE, and thank you for providing a screen video demonstrating the functionality.  (This makes PX review much easier.)
Looks great.

**Comment by yapei** [comment]
@jcaianirh it looks like we still have some issues on following pages
1. API Explorer page, no API resources can be found even searching string matches the Kind
see my recording
https://github.com/openshift/console/assets/12692381/e9f47807-7504-4197-adcb-e5725b917fa9
2. /settings/cluster/globalconfig page doesn't support exact match yet

**Comment by jcaianirh** [comment]
> @jcaianirh it looks like we still have some issues on following pages
> 
> 1. API Explorer page, no API resources can be found even searching string matches the Kind
>    see my recording
>    https://github.com/openshift/console/assets/12692381/e9f47807-7504-4197-adcb-e5725b917fa9
> 2. /settings/cluster/globalconfig page doesn't support exact match yet

@yapei you have the exact match setting enabled so typing config will not search for Config... you need an exact match so the lower case c will not fetch the upper case Config

**Comment by yapei** [comment]
@jcaianirh you are right ~ I tried again and exact match is working well on `API Explorer` page
how about this issue? 
```
/settings/cluster/globalconfig page doesn't support exact match yet
```

**Comment by jcaianirh** [comment]
> @jcaianirh you are right ~ I tried again and exact match is working well on `API Explorer` page how about this issue?
> 
> ```
> /settings/cluster/globalconfig page doesn't support exact match yet
> ```

@yapei I think this page is ok. this is more for list pages. there are no user generated instances here so i think searching with fuzzy is ok.

**Comment by jcaianirh** [comment]
> @jcaianirh you are right ~ I tried again and exact match is working well on `API Explorer` page how about this issue?
> 
> ```
> /settings/cluster/globalconfig page doesn't support exact match yet
> ```

@yapei I think this page is ok. this is more for list pages. there are no user generated instances here so i think searching with fuzzy is ok plus the search also looks at description as well. I don't think the strict/fuzzy will make an impact on results here.

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**
